core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -1142,20 +1144,10 @@ private[spark] class AppStatusListener(
       s.info.status != v1.StageStatus.ACTIVE && s.info.status != v1.StageStatus.PENDING
     }
 
-    stages.foreach { s =>
+    val stageIndexValues = stages.map { s =>
```

Comment:
`s/stageIndexValues/stageIds` (or `stageKeys`, since here they're actually the primary keys of the stages being removed...)